src/metabase/sync/interface.clj

```diff
@@ -97,6 +97,40 @@
 (def Fingerprint
   "Schema for a Field 'fingerprint' generated as part of the analysis stage. Used to power the 'classification' sub-stage of
    analysis. Stored as the `fingerprint` column of Field."
-  {(s/optional-key :global)       GlobalFingerprint
+  {(s/optional-key :version)      su/IntGreaterThanZero ; Fingerprints with no version key are assumed to have version of 1
```

Comment:
If I'm understanding this correctly, we won't be able to make the decision as to where or not we need to "upgrade" a fingerprint until after we have read the row, parsed the JSON and examined the version. That seems like it could impact performance pretty significantly.